### PR TITLE
Check the file is accessible before trying to read it

### LINKED ISSUE #8: Check for file access before trying to read it
Access-denied and file-not-found probably manifest as decryption failure errors right now.

### PR DESCRIPTION
**Pull request by elefeint**
I really should have fixed it the first time I lost time troubleshooting decryption.
Also makes `IS_FAIL` both more and less flexible -- it now supports substring matching rather than full string, but it also fails if the criterion is not matched.

Fixes #8

### CODE REVIEW

File: test/integration/test_server.cpp [hunk 1]
```diff
@@ -24,10 +24,13 @@ bool NO_FAIL(const grpc::Status &status) {
   return status.ok();
 }
 
-bool IS_FAIL(const grpc::Status &status, const std::string &expected_error) {
-  if (!status.ok() && status.error_message() != expected_error) {
-    fprintf(stderr, "Query failed with unexpected message: %s\n",
-            status.error_message().c_str());
+bool IS_FAIL(const grpc::Status &status,
```
Comment (guenp) [inline test/integration/test_server.cpp:27]: How is this test related to the change?

File: test/integration/test_server.cpp [hunk 2]
```diff
@@ -818,4 +821,27 @@ TEST_CASE("Truncate fails if synced_column is missing") {
...
+  request.mutable_csv()->set_encryption(::fivetran_sdk::Encryption::AES);
+  request.mutable_csv()->set_compression(::fivetran_sdk::Compression::ZSTD);
+  define_test_table(request, "unused_table");
+
+  request.add_replace_files(bad_file_name);
+  (*request.mutable_keys())[bad_file_name] = "whatever";
+
+  ::fivetran_sdk::WriteBatchResponse response;
+  auto status = service.WriteBatch(nullptr, &request, &response);
+  REQUIRE_FAIL(status, "is missing or inaccessible");
```
Comment (guenp) [inline test/integration/test_server.cpp:846]: Would this work?
```suggestion
  REQUIRE_FAIL(status, "File <" + TEST_RESOURCES_DIR + "nonexistent.csv> is missing or inaccessible");
```

**Review by guenp — APPROVED** [review]
Approved with suggestions

File: test/integration/test_server.cpp (hunk 1 above)
Comment (elefeint) [reply test/integration/test_server.cpp:27]: I tried to use REQUIRE_FAIL to test the change and realized the behavior in IS_FAIL was not useful.

File: test/integration/test_server.cpp (hunk 2 above)
Comment (elefeint) [reply test/integration/test_server.cpp:846]: yep, I'll do that

File: src/motherduck_destination_server.cpp [hunk 3]
```diff
@@ -85,12 +86,23 @@ get_encryption_key(const std::string &filename,
   return encryption_key_it->second;
 }
 
+void validate_file(const std::string &filename) {
```
[the remark this answers is not in the document]
Comment (elefeint) [reply src/motherduck_destination_server.cpp:89]: indeed

File: test/integration/test_server.cpp (hunk 1 above)
Comment (elefeint) [reply test/integration/test_server.cpp]: I put the strict equality check back, but made it a native Catch2 check, so it prints the diffs correctly. I did keep the new-to-this-PR behavior of failing if the message is wrong.
